irc_server.go

```diff
@@ -390,6 +391,55 @@ func IrcPassHandler(ctx *IrcContext, prefix, cmd string, args []string, trailing
 	ctx.SlackAPIKey = args[0]
 }
 
+// IrcTopicHandler is called when a TOPIC command (TOPIC <channel> [<topic>]) is sent
+func IrcTopicHandler(ctx *IrcContext, prefix, cmd string, args []string, trailing string) {
+	if len(args) < 1 {
+		// ERR_NEEDMOREPARAMS
+		SendIrcNumeric(ctx, 461, ctx.Nick(), "TOPIC :Not enough parameters")
+		return
+	}
+	channame := args[0]
+	if strings.HasPrefix(channame, "#") {
+		channame = channame[1:]
+	}
+	chantopic := trailing
+	// Disallow a user to change the topic of a channel
+	// for which the user isn't a member.
+	if _, found := ctx.Channels[channame]; !found {
+		// ERR_NOTONCHANNEL
+		SendIrcNumeric(ctx, 442, ctx.Nick(), fmt.Sprintf("%s :You're not on that channel", args[0]))
+		return
+	}
+	// TODO Check if user requires 'chanop' privileges to change the topic (ERR_CHANOPRIVSNEEDED)
+
+	// Slack needs the channel ID to set the topic, not the channel name. The only
```

Comment:
the code below probably deserves its own function, as it is generic enough. Can you split it out and put it in `irc_context.go`? This would pair with `IrcContext.GetUser*`